app/_src/policies/hostnamegenerator.md

```diff
@@ -0,0 +1,72 @@
+---
+title: HostnameGenerator
+---
+
+{% warning %}
+This resource is experimental!
+{% endwarning %}
+
+A `HostnameGenerator` provides
+
+- a template to generate hostnames from properties of `MeshServices` and `MeshExternalServices`
+- a selector that defines for which `MeshServices` and `MeshExternalServices` this generator runs
+
+{% policy_yaml hg-all %}
+```yaml
+type: HostnameGenerator
+name: all
+spec:
+  selector:
+    meshService:
+      matchLabels:
+        k8s.kuma.io/namespace: kuma-demo
+  template: "{{ .DisplayName }}.{{ .Namespace }}.mesh"
+```
+{% endpolicy_yaml %}
+
+## Template
+
+A template is a [golang text template](https://pkg.go.dev/text/template).
+It is run with the name, namespace and mesh as input and function `label` to retrieve labels of the `MeshService` or `MeshExternalService`.
+
+For exampe, given:
+
+```yaml
+kind: MeshService
+metadata:
+  name: redis
+  namespace: kuma-demo
+  labels:
+    kuma.io/mesh: products
+    team: backend
+    k8s.kuma.io/service-name: redis
+    k8s.kuma.io/namespace: kuma-demo
+```
+
+and
+
+```
+  template: "{{ .DisplayName }}.{{ .Namespace }}.{{ .Mesh }}.{{ label "team" }}.mesh"
```

Comment:
We meed to list every field with explanation. I mean that display name is original name before the object is synced around, on kube this is just a name. We also should say that namespace is original namespace of synced resource